File: backend/Models/Rep.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace backend.Models;

[Table("Reps")]
public class Rep
{
    [Key]
    public int RepID { get; set; }

    [Required]
    public int SessionID { get; set; }

    [Required]
    public int RepNumber { get; set; }

    public float? RepScore { get; set; }

    [ForeignKey("SessionID")]
    public Sessions? Session { get; set; }

    // Navigation property for frames
    public ICollection<Frame> Frames { get; set; } = new List<Frame>();
}

File: frontend/src/types/rep.ts
import type { CreateFrameRequest, FrameResponse } from "./frame";

export interface RepResponse {
  repID: number;
  repNumber: number;
  repScore: number;
  frames?: FrameResponse[];
}

export interface CreateRepRequest {
  repNumber: number;
  repScore?: number;
  frames?: CreateFrameRequest[];
}
